### ISSUE TITLE
Validation: wildcard symbol to match any one level of the array not working correctly 

### ISSUE THREAD

**Issue by jahmedk** [codeigniter4/CodeIgniter4#5286]
### PHP Version

7.3

### CodeIgniter4 Version

4.0.5

### Which operating systems have you tested for this bug?

Windows

### Which server did you use?

apache

### Database

mysql

### What happened?

One level array validating just 0 index but it's not working on multiple records.

### Steps to Reproduce

![image](https://user-images.githubusercontent.com/22505794/140501415-4c648840-4c8e-4f3c-aed3-26a63db15cf8.png)
![image](https://user-images.githubusercontent.com/22505794/140501128-0a1236e5-a617-4c75-88df-0fc18af8f852.png)

![image](https://user-images.githubusercontent.com/22505794/140501165-f312b63c-e058-43b4-b839-8da7a38c5425.png)
![image](https://user-images.githubusercontent.com/22505794/140501200-792c392d-b738-446c-80a3-3995b71116e3.png)

### Expected Output

$validation->run($aVars) must be false if array contain invalid data

**Comment by kenjis** [comment]
Can't reproduce on the latest `develop` branch.
It seems it has been fixed.

`tests/system/Validation/ValidationTest.php`:
```php
    public function test_5286(): void
    {
        $rules = [
            'arl1.*.id' => 'required|regex_match[/^[0-9]+$/]',
            'arl1.*.alpha' => 'required|regex_match[/^[?a-zA-Z ]+$/]',
        ];
        $data = [
            'arl1' => [
                [
                    'id' => 101,
                    'alpha' => 'Simple String',
                ],
                [
                    'id' => '#10003',
                    'alpha' => 'Simple & Symbol',
                ],
            ],
        ];
        $actual = $this->validation->setRules($rules)->run($data);
        d($actual);
        dd($this->validation->getErrors());
    }
```

```sh-session
$ ./phpunit tests/system/Validation/ValidationTest.php --filter=test_5286
PHPUnit 9.5.10 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.0.12
Configuration: /Users/kenji/work/codeigniter/CodeIgniter4/phpunit.xml


┌─────────────────────────────────────────────────────────────────────────────────────────────────┐
│ $actual                                                                                         │
└─────────────────────────────────────────────────────────────────────────────────────────────────┘
boolean false
═══════════════════════════════════════════════════════════════════════════════════════════════════
Called from <ROOT>/tests/system/Validation/ValidationTest.php:1032 [d()]

┌─────────────────────────────────────────────────────────────────────────────────────────────────┐
│ $this->validation->getErrors()                                                                  │
└─────────────────────────────────────────────────────────────────────────────────────────────────┘
array (2) [
    'arl1.*.id' => string (49) "The arl1.*.id field is not in the correct format."
    'arl1.*.alpha' => string (52) "The arl1.*.alpha field is not in the correct format."
]
═══════════════════════════════════════════════════════════════════════════════════════════════════
Called from <ROOT>/tests/system/Validation/ValidationTest.php:1033 [dd()]
```

**Comment by kenjis** [comment]
Duplicate of #4510
Fixed by #4606

@jahmedk Can you upgrade to v4.1.4?
This is fixed in v4.1.2.

**Comment by jahmedk** [comment]
Hi yes
Thanks @kenjis
